step/src/main/java/com/google/sps/Notification.java

```diff
@@ -15,5 +15,56 @@
 package com.google.sps;
 
 public abstract class Notification {
-  // TODO: implement this class    
-}
\ No newline at end of file
+  private long id;  
+  protected String otherUser;
+  private long timestamp;
+
+  public Notification(long id, String otherUser, long timestamp) {
+    this.id = id;
+    this.otherUser = otherUser;
+    this.timestamp = timestamp;
+  }
+  
+  // Getter method for the ID of the user with whom the notification is
+  // associated.
+  public long getId() {
+    return id;    
+  }
+
+  // Method for getting the name of the user who sent the notification.
+  public String getOtherUser() {
+    return otherUser;  
+  }
+  
+  // Getter method for the time at which the notification occured.
+  public long getTimestamp() {
+    return timestamp;  
+  }
+
+  // Overriden equals method
+  @Override
+  public boolean equals(Object obj) { 
+    if(this == obj) {
+      return true; 
+    } 
+ 
+    if(obj == null || obj.getClass()!= this.getClass()) {
+      return false;
+    }
+     
+    Notification notification = (Notification) obj;
+    return this.id == notification.getId() &&
+           this.otherUser.equals(notification.getOtherUser()) &&
+           this.timestamp == notification.getTimestamp(); 
+  }
+
+  // Overriden hashCode method
+  @Override
+  public int hashCode() { 
+    return (int) id * otherUser.hashCode() * (int) timestamp; 
+  }
```

Comment:
I don't think the hashcode needs to be overwritten since Object.hashCode() should already be producing a unique hash. Additionally, I think your hash algorithm could produce an int overflow since System.currentTimeMillis() returns pretty big numbers (it returns the number of milliseconds since 1/1/1970.

Reply:
Any class that overrides the equals method needs to also override hashCode otherwise you can run into some funky errors when using hashMaps or other structures that rely on hashing. I can definitely get rid of currentTimeMillis though.